pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -447,19 +457,42 @@ func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *v1.Pod, podS
 	// check the status of containers.
 	for index, container := range pod.Spec.Containers {
 		containerStatus := podStatus.FindContainerStatusByName(container.Name)
-		if containerStatus == nil || containerStatus.State != kubecontainer.ContainerStateRunning {
+		containerRef, err := kubecontainer.GenerateContainerRef(pod, &container)
+		if err != nil {
+			glog.Errorf("Couldn't make a ref to pod %q, container %v: '%v'", format.Pod(pod), container.Name, err)
+		}
+
+		var recordedContainerState v1.ContainerState
+		if len(existingStatus.ContainerStatuses) > index {
+			recordedContainerState = existingStatus.ContainerStatuses[index].State
```

Comment:
Definitely need to fix this... I should be finding the container by name instead, but I got lost looking for the helper function for this that accepts a v1.PodStatus as opposed to a kubecontainer.PodStatus...